### PR TITLE
Complete metadata query

### PR DESCRIPTION
**Pull request by garrettmflynn**
This PR introduces frontend and backend code to handle a complete metadata query. It renders minimal metadata results from `neuroconv` (e.g. required properties) on the Metadata page.

### CODE REVIEW

File: pyflask/manageNeuroconv/manage_neuroconv.py [hunk 1]
```diff
@@ -28,20 +28,30 @@ def get_all_interface_info() -> dict:
...
-
-    # Combine Multiple Interfaces
-    class CustomNWBConverter(NWBConverter):
-        data_interface_classes = {interface: getattr(datainterfaces, interface) for interface in interface_class_names}
-
+    CustomNWBConverter = get_custom_converter(interface_class_names)
     return CustomNWBConverter.get_source_schema()
 
 
 def get_metadata(source_data):
```
Comment (bendichter) [inline pyflask/manageNeuroconv/manage_neuroconv.py:47]:
```suggestion
def get_metadata_schema(source_data):
```

Comment (garrettmflynn) [reply pyflask/manageNeuroconv/manage_neuroconv.py]: Updated all cases of this function in a subsequent push

File: pyflask/manageNeuroconv/manage_neuroconv.py [hunk 2]
```diff
@@ -28,20 +28,30 @@ def get_all_interface_info() -> dict:
...
+    """
+    CustomNWBConverter = get_custom_converter(interface_class_names)
     return CustomNWBConverter.get_source_schema()
 
 
-def get_metadata(source_data):
+def get_metadata_schema(source_data):
     """
-    Function used to get metadata from a CustomNWBConverter
+    Function used to get metadata schea from a CustomNWBConverter
```
Comment (CodyCBakerPhD) [inline pyflask/manageNeuroconv/manage_neuroconv.py:49]:
```suggestion
    Function used to fetch the metadata schema from a CustomNWBConverter instantiated from the source_data.
```

**Comment by CodyCBakerPhD** [comment]
One thing to note that I found here is the last field of the subject metadata clips below the bottom bar

image

If you can fix that in a follow-up, great, but it's functional for now

File: pyflask/manageNeuroconv/manage_neuroconv.py [hunk 3]
```diff
@@ -28,20 +31,32 @@ def get_all_interface_info() -> dict:
...
     """
-    Function used to get metadata from a CustomNWBConverter
+    Function used to fetch the metadata schema from a CustomNWBConverter instantiated from the source_data.
     """
-    return source_data  # Test function by echoing back the input
+
+    interface_class_names = list(
+        source_data
+    )  # NOTE: We currently assume that the keys of the properties dict are the interface names
+    CustomNWBConverter = get_custom_converter(interface_class_names)
```
Comment (CodyCBakerPhD) [inline pyflask/manageNeuroconv/manage_neuroconv.py:58]: Yes, we will have to think of a better way of managing this going forward, but I have some ideas and we'll probably want to have a specific design discussion (either an issue or on Zoom) to decide what's best, but the in the race for MVP it's fine for now

File: pyflask/manageNeuroconv/manage_neuroconv.py [hunk 4]
```diff
@@ -28,20 +31,32 @@ def get_all_interface_info() -> dict:
...
+def get_source_schema(interface_class_names: List[str]) -> dict:
+    """
+    Function used to get schema from a CustomNWBConverter that can handle multiple interfaces
+    """
+    CustomNWBConverter = get_custom_converter(interface_class_names)
     return CustomNWBConverter.get_source_schema()
 
 
-def get_metadata(source_data):
+def get_metadata_schema(source_data):
```
Comment (CodyCBakerPhD) [inline pyflask/manageNeuroconv/manage_neuroconv.py:50]: What is the `->` return type here?

File: pyflask/manageNeuroconv/manage_neuroconv.py [hunk 5]
```diff
@@ -28,20 +31,32 @@ def get_all_interface_info() -> dict:
...
-    return source_data  # Test function by echoing back the input
+
+    interface_class_names = list(
+        source_data
+    )  # NOTE: We currently assume that the keys of the properties dict are the interface names
+    CustomNWBConverter = get_custom_converter(interface_class_names)
+    converter = CustomNWBConverter(source_data)
+    schema = converter.get_metadata_schema()
+    metadata = converter.get_metadata()
+    return json.loads(json.dumps(dict(results=metadata, schema=schema), cls=NWBMetaDataEncoder))
```
Comment (CodyCBakerPhD) [inline pyflask/manageNeuroconv/manage_neuroconv.py:62]: Why a second `json.loads` here? (also plays into understanding what the return type here is)

File: pyflask/manageNeuroconv/manage_neuroconv.py (hunk 3 above)
Comment (garrettmflynn) [reply pyflask/manageNeuroconv/manage_neuroconv.py:58]: Sounds good!

**Comment by garrettmflynn** [comment]
Thank you for the feedback! Yeah, I'll do some UI cleanup in a separate PR since there is quite a bit lingering in the background.

File: pyflask/manageNeuroconv/manage_neuroconv.py (hunk 5 above)
Comment (garrettmflynn) [reply pyflask/manageNeuroconv/manage_neuroconv.py:62]: I was able to confirm that I cannot directly send the `json.dumps` output using Flask

Comment (CodyCBakerPhD) [reply pyflask/manageNeuroconv/manage_neuroconv.py:62]: I see, thanks for confirming!

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**